Packs/CommonScripts/Scripts/ParseEmailFiles/ParseEmailFiles.py

```diff
@@ -3387,27 +3387,47 @@ def get_utf_string(text, field):
-    if encoding.lower() == 'b':
-        byte_string = base64.b64decode(encoded_text)
-    elif encoding.lower() == 'q':
-        byte_string = quopri.decodestring(encoded_text)
-    return prefix + byte_string.decode(charset) + suffix
+def mime_decode(encoded_string):
+    word_mime_encoded = MIME_ENCODED_WORD.search(encoded_string)
+    if word_mime_encoded:
+        prefix, charset, encoding, encoded_text, suffix = word_mime_encoded.groups()
+        if encoding.lower() == 'b':
+            byte_string = base64.b64decode(encoded_text)
+        elif encoding.lower() == 'q':
+            byte_string = quopri.decodestring(encoded_text)
+        try:
+            return prefix + byte_string.decode(charset) + suffix
+        except UnicodeDecodeError:
+            demisto.debug('Failed to decode encoded_string: {}. charset: {}, '
+                          'encoding: {}, encoded_text: {}'.format(encoded_string, charset, encoding, encoded_text))
+            return prefix + byte_string.decode(charset, errors='replace') + suffix
+    return ''
 
 
 def convert_to_unicode(s, is_msg_header=True):
     global ENCODINGS_TYPES
     try:
         res = ''  # utf encoded result
         if is_msg_header:  # Mime encoded words used on message headers only
+            encode_decode_phrase = s
             try:
-                word_mime_encoded = s and MIME_ENCODED_WORD.search(s)
+                word_mime_encoded = MIME_ENCODED_WORD.search(encode_decode_phrase)
                 if word_mime_encoded:
-                    word_mime_decoded = mime_decode(word_mime_encoded)
-                    if word_mime_decoded and not MIME_ENCODED_WORD.search(word_mime_decoded):
-                        # ensure decoding was successful
-                        return word_mime_decoded
+                    if '?= =?' in encode_decode_phrase:
+                        encode_decode_phrase = encode_decode_phrase.replace("?= =?", '?==?')
+                    while word_mime_encoded:
+                        # encoded-word" is a sequence of printable ASCII characters that begins with "=?",
+                        # ends with "?=", and has two "?"s in between.
+                        start_encoding_index = encode_decode_phrase.index("=?")
+                        end_encoding_index = encode_decode_phrase.index("?=") + 2
+                        # index return the index where "?=" starts, need to include it on the substring
+                        encoded_substring = encode_decode_phrase[start_encoding_index:end_encoding_index]
+                        decoded_substring = mime_decode(encoded_substring)
+                        encode_decode_phrase = encode_decode_phrase[:start_encoding_index] + decoded_substring + \
+                                               encode_decode_phrase[end_encoding_index:]  # noqa: E127
+                        word_mime_encoded = MIME_ENCODED_WORD.search(encode_decode_phrase)
+                    return encode_decode_phrase
```

Comment:
looks good, please just fix the mix of use in in " and '. use only ' for strings.

Reply:
It is ready for merge